Plutarch/Api/V1/Value.hs

```diff
@@ -38,13 +54,200 @@ deriving via
   instance
     PConstantDecl Plutus.CurrencySymbol
 
-newtype PValue (s :: S) = PValue (Term s (PMap PCurrencySymbol (PMap PTokenName PInteger)))
+data ValueKeyGuarantees = Sorted | Unsorted
+data ValueAmountGuarantees = NoGuarantees | NonZero | Positive
+
+newtype PValue (keys :: ValueKeyGuarantees) (amounts :: ValueAmountGuarantees) (s :: S)
```

Comment:
So we have some subtyping relations here, I wonder whether we should add `PSubtype` instances. `PSubtype` essentially gives us one-way coercions.

Reply:
Do you mean you want to change the current definition of `PSubtypeDecl`? It's not overridable as it stands.

Reply:
Yep.